pkg/operator/ceph/controller/controller_utils.go

```diff
@@ -138,18 +139,19 @@ func EnforceHostNetwork() bool {
 }
 
 func SetRevisionHistoryLimit(data map[string]string) {
-	strval := k8sutil.GetValue(data, revisionHistoryLimitSettingName, revisionHistoryLimitDefaultValue)
-	if strval != "" {
+	strval := k8sutil.GetValue(data, revisionHistoryLimitSettingName, revisionHistoryLimitDefaultStr)
+	var limit int32
+	if strval == "" {
+		limit = revisionHistoryLimitDefault
+	} else {
 		numval, err := strconv.ParseInt(strval, 10, 32)
 		if err != nil {
-			logger.Warningf("failed to parse value %q for %q. assuming default value.", strval, revisionHistoryLimitSettingName)
-			revisionHistoryLimit = nil
```

Comment:
Why can't we set `nil` when the value isn't set or doesn't parse correctly?

Reply:
> Why can't we set `nil` when the value isn't set or doesn't parse correctly?

Because  using `RevisionHistoryLimit()`in DeploymentSpecs for `RevisionHistoryLimit:`whinch expexts `*int32` thewn results in nil pointer deref.

Reply:
https://go.dev/play/p/e5p6BsWdMBN

Reply:
of course we can set revisionHistoryLimit to `nil`, but some code paths dereference it and that leads to problems.